src/components/TxConfirmDialog/index.js

```diff
@@ -2,44 +2,104 @@ import React, { Component } from 'react';
 import Dialog, { DialogTitle, DialogContent, DialogActions } from 'material-ui/Dialog';
 import { FormattedMessage, injectIntl, intlShape, defineMessages } from 'react-intl';
+import Table, {
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  TableFooter,
+} from 'material-ui/Table';
 
 import appActions from '../../redux/App/actions';
+import styles from './styles';
 
 const messages = defineMessages({
   confirmMessage: {
     id: 'txConfirm.message',
     defaultMessage: 'You are about to {txDesc} for {txAmount} {txToken}. Please click OK to continue.',
   },
+  confirmMessageWithFee: {
+    id: 'txConfirm.messageWithFee',
+    defaultMessage: 'You are about to {txDesc} for {txAmount} {txToken}, with fee for {txFee} QTUM. Please click OK to continue.',
+  },
 });
 
 @injectIntl
+@withStyles(styles, { withTheme: true })
 @connect((state) => ({
   txConfirmInfoAndCallback: state.App.get('txConfirmInfoAndCallback'),
+  transactionCost: state.App.get('transactionCost'),
 }), (dispatch) => ({
   clearTxConfirm: () => dispatch(appActions.clearTxConfirm()),
+  getTransactionCost: (txInfo) => dispatch(appActions.getTransactionCost(txInfo)),
 }))
+
 export default class TxConfirmDialog extends Component {
   static propTypes = {
+    classes: PropTypes.object.isRequired,
     intl: intlShape.isRequired, // eslint-disable-line react/no-typos
     txConfirmInfoAndCallback: PropTypes.object.isRequired,
     clearTxConfirm: PropTypes.func.isRequired,
+    transactionCost: PropTypes.array.isRequired,
+    getTransactionCost: PropTypes.func.isRequired,
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { txConfirmInfoAndCallback, getTransactionCost } = nextProps;
+    const { txInfo } = txConfirmInfoAndCallback;
+
+    if (txConfirmInfoAndCallback !== this.props.txConfirmInfoAndCallback && txInfo) {
+      getTransactionCost(txInfo);
+    }
   }
 
   render() {
-    const { intl: { formatMessage }, txConfirmInfoAndCallback } = this.props;
-    const { txDesc, txAmount, txToken, confirmCallback } = txConfirmInfoAndCallback;
+    const { classes, intl: { formatMessage }, txConfirmInfoAndCallback, transactionCost } = this.props;
+    const { txDesc, txAmount, txToken, txInfo, confirmCallback } = txConfirmInfoAndCallback;
     const isOpen = !!(txDesc && txAmount && txToken && _.isFunction(confirmCallback));
+    const txFee = _.sumBy(transactionCost, (cost) => cost.gasCost ? parseFloat(cost.gasCost) : 0);
 
     return (
       <Dialog open={isOpen}>
         <DialogTitle>
           <FormattedMessage id="txConfirm.title" defaultMessage="Please Confirm Your Transaction" />
         </DialogTitle>
         <DialogContent>
-          {formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken })}
+          { txFee ? formatMessage(messages.confirmMessageWithFee, { txDesc, txAmount, txToken, txFee }) : formatMessage(messages.confirmMessage, { txDesc, txAmount, txToken }) }
+          { transactionCost.length ?
```

Comment:
1. same as above
2. please put parenthesis on same line
3. please don't use ternaries in places like this
`{!!transactionCost.length && (`
please see [React Best Practices in frontify](https://bodhi.frontify.com/document/230177#/react-best-practices)